c7n/resources/eks.py

```diff
@@ -146,7 +146,54 @@ class Delete(Action):
     def process(self, resources):
         client = local_session(self.manager.session_factory).client('eks')
         for r in resources:
+            nodegroups = client.list_nodegroups(clusterName=r['name'])['nodegroups']
```

Comment:
this whole thing should be in a separate function with a try/except not found around the method usage.

Reply:
addressed in 7c656a5